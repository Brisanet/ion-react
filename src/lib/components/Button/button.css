.btn {
  padding: 0.5rem 1rem;
  border: 1px solid lightblue;
  border-radius: 0.5rem;
  font-weight: bold;
  font-size: 1rem;
  color: #444;
  background-color: lightblue;
}
