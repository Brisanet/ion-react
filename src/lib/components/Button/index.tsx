import "./button.css";

interface ButtonInterface {
  children: string | JSX.Element;
}

const Button = ({ children }: ButtonInterface): JSX.Element => {
  return <button className="btn">{children}</button>;
};

export default Button;
