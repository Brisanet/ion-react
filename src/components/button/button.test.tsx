import React from 'react';
import { render } from '@testing-library/react';

import Button from './button';

describe('Button', () => {
  test('renders the Button component', () => {
    render(<Button label="Hello world!" />);
  });
});
