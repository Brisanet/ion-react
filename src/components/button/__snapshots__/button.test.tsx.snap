// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button Sizes should render button with lg size variation 1`] = `
.c0 {
  appearance: none;
  font-family: Source Sans Pro,sans-serif;
  font-weight: 600;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0px;
  flex-direction: row;
  height: 40px;
  min-width: 40px;
  padding: 8px 20px;
  font-size: 1.6rem;
  line-height: 2.4rem;
  background-color: #0858ce;
  color: #fcfcfd;
  border: none;
}

.c0 svg {
  fill: #fcfcfd;
}

.c0:hover,
.c0:focus-visible {
  background-color: #146ff5;
  border: none;
}

.c0:active {
  background-color: #06439d;
  border: none;
}

.c0:disabled {
  background-color: #e4e6eb;
  color: #aeb2bd;
  border: none;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

<button
  class="c0"
  data-testid="ion-button"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button Button Sizes should render button with md size variation 1`] = `
.c0 {
  appearance: none;
  font-family: Source Sans Pro,sans-serif;
  font-weight: 600;
  border-radius: 6px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0px;
  flex-direction: row;
  height: 32px;
  min-width: 32px;
  padding: 6px 16px;
  font-size: 1.4rem;
  line-height: 2rem;
  background-color: #0858ce;
  color: #fcfcfd;
  border: none;
}

.c0 svg {
  fill: #fcfcfd;
}

.c0:hover,
.c0:focus-visible {
  background-color: #146ff5;
  border: none;
}

.c0:active {
  background-color: #06439d;
  border: none;
}

.c0:disabled {
  background-color: #e4e6eb;
  color: #aeb2bd;
  border: none;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

<button
  class="c0"
  data-testid="ion-button"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button Button Sizes should render button with sm size variation 1`] = `
.c0 {
  appearance: none;
  font-family: Source Sans Pro,sans-serif;
  font-weight: 600;
  border-radius: 6px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0px;
  flex-direction: row;
  height: 24px;
  min-width: 24px;
  padding: 4px 12px;
  font-size: 1.2rem;
  line-height: 1.6rem;
  background-color: #0858ce;
  color: #fcfcfd;
  border: none;
}

.c0 svg {
  fill: #fcfcfd;
}

.c0:hover,
.c0:focus-visible {
  background-color: #146ff5;
  border: none;
}

.c0:active {
  background-color: #06439d;
  border: none;
}

.c0:disabled {
  background-color: #e4e6eb;
  color: #aeb2bd;
  border: none;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

<button
  class="c0"
  data-testid="ion-button"
>
  <span>
    Button
  </span>
</button>
`;

exports[`Button Button Sizes should render button with xl size variation 1`] = `
.c0 {
  appearance: none;
  font-family: Source Sans Pro,sans-serif;
  font-weight: 600;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0px;
  flex-direction: row;
  height: 48px;
  min-width: 48px;
  padding: 12px 24px;
  font-size: 1.6rem;
  line-height: 2.4rem;
  background-color: #0858ce;
  color: #fcfcfd;
  border: none;
}

.c0 svg {
  fill: #fcfcfd;
}

.c0:hover,
.c0:focus-visible {
  background-color: #146ff5;
  border: none;
}

.c0:active {
  background-color: #06439d;
  border: none;
}

.c0:disabled {
  background-color: #e4e6eb;
  color: #aeb2bd;
  border: none;
  cursor: not-allowed;
}

.c0:disabled svg {
  fill: #aeb2bd;
}

.c0:focus-visible {
  outline: 2px solid #146ff5;
  outline-offset: 2px;
}

<button
  class="c0"
  data-testid="ion-button"
>
  <span>
    Button
  </span>
</button>
`;
